packages/migrations/src/batched-migrations/batched-migration.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  loadSqlEquiv,
+  queryAsync,
+  queryValidatedOneRow,
+  queryValidatedRows,
+  queryValidatedZeroOrOneRow,
+} from '@prairielearn/postgres';
+import { z } from 'zod';
+
+const sql = loadSqlEquiv(__filename);
+
+export const BatchedMigrationStatusSchema = z.enum([
+  'pending',
+  'paused',
+  'running',
+  'finalizing',
+  'failed',
+  'succeeded',
+]);
+export type BatchedMigrationStatus = z.infer<typeof BatchedMigrationStatusSchema>;
+
+export const BatchedMigrationRowSchema = z.object({
+  id: z.string(),
+  project: z.string(),
+  filename: z.string(),
+  timestamp: z.string(),
+  batch_size: z.number(),
+  min_value: z.bigint({ coerce: true }),
+  max_value: z.bigint({ coerce: true }),
+  status: BatchedMigrationStatusSchema,
+  created_at: z.date(),
+  updated_at: z.date(),
+  started_at: z.date().nullable(),
+});
+export type BatchedMigrationRow = z.infer<typeof BatchedMigrationRowSchema>;
+
+export interface BatchedMigrationParameters {
+  min?: bigint | null;
+  max: bigint | null;
+  batchSize?: number;
+}
+
+export interface BatchedMigrationImplementation {
+  getParameters(): Promise<BatchedMigrationParameters>;
+  execute(start: bigint, end: bigint): Promise<void>;
+}
```

Comment:
In short: because "batched migration" throughout this code means both "a row in the `batched_migrations` table" and "the code for a specific batched migration" and I wanted the naming to reflect that fact. I'm open to changing that, but maybe there's a better name than just `BatchedMigration`?

Reply:
That makes enough sense to me.